libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -1919,17 +1916,22 @@ class MapboxRouteLineApi(
         alternativelyStyleSegmentsNotInLegCache.evictAll()
     }
 
-    internal val alternativelyStyleSegmentsNotInLeg: (
+    private val alternativelyStyleSegmentsNotInLeg: (
         Int,
         List<RouteLineExpressionData>,
-        Int
+        InactiveRouteColors
     ) -> List<RouteLineExpressionData> =
-        { activeLegIndex: Int, segments: List<RouteLineExpressionData>, substitutionColor: Int ->
+        { activeLegIndex: Int, segments: List<RouteLineExpressionData>, _: InactiveRouteColors ->
             segments.parallelMap(
                 {
                     if (it.legIndex != activeLegIndex) {
                         it.copyWithNewSegmentColor(
-                            newSegmentColor = substitutionColor
+                            newSegmentColor = getRouteColorForCongestion(
+                                congestionValue = it.congestionValue,
+                                isPrimaryRoute = true,
+                                isInactive = true,
+                                routeLineOptions.resourceProvider.routeLineColorResources
```

Comment:
Shouldn't you be using `InactiveRouteColors` here? I don't see you use it anywhere actually, except for creation.

Reply:
I'm using `InactiveRouteColors` only as a cache key for the function that calculates the segments when legs are inactive. I didn't want to use the whole `MapboxRouteLineOptions` but using `RouteLineColorResources` should be fine as well, these are only `Int`s.